### PR TITLE
Create 5e_lay_on_hands.js

### PR DESCRIPTION
**Pull request by aka-beer-buddy**
Gives the user the ability to define how many HP of healing to apply to a target character. If the macro has permissions, it will apply the HP; otherwise, it will produce a Roll for the target user to apply HP heal manually.

### CODE REVIEW

File: actor/5e_lay_on_hands.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/**
+ * System: D&D5e
+ * Apply lay-on-hands feat to a target character.  Asks the player how many HP to heal and
+ * verifies the entered value is within range before marking down usage counter. If the player
+ * has OWNER permissions of target (such as GM or self-heal) the HP are applied automatically; 
+ * otherwise, a 'roll' message appears allowing the target character to right-click to apply healing.
+ */
+
+let confirmed = false;
+let actorData = actor ? actor : canvas.tokens.controlled[0];
```
Comment (Merudo) [inline actor/5e_lay_on_hands.js:10]: We could try the default character if no token is controlled:

```suggestion
let actorData = actor ? actor : canvas.tokens.controlled[0] || game.user.character;;
```

**Comment by Merudo** [comment]
I tried the macro but it wastes 1 charge every time it is launched.

**Comment by johnsandberg** [comment]
> I tried the macro but it wastes 1 charge every time it is launched.

I was not able to recreate this either as GM or as player.  There are no charges used unless the user enters a value and selects "heal!".  I hope this isn't a module conflict.  I do see a message needs to be improved to tell player to manually apply healing when the macro user doesn't have permissions.  I'll add that in.

File: actor/5e_lay_on_hands.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    ui.notifications.warn(`Invalid number of charges entered = ${number}. Aborting action.`);
+                else
+                {
+                    let flavor = `<strong>${html.find('#flavor')[0].value}</strong><br>`;
+                    if (targetActor.permission !== CONST.ENTITY_PERMISSIONS.OWNER)
+                        // We need help applying the healing, so make a roll message for right-click convenience.
+                        new Roll(`${number}`).roll().toMessage({
+                            speaker: ChatMessage.getSpeaker(),
+                            flavor: `${actorData.name} lays hands on ${targetActor.data.name}.<br>${flavor}` });
+                    else {
```
Comment (johnsandberg) [inline actor/5e_lay_on_hands.js:59]:
```suggestion
                            flavor: `${actorData.name} lays hands on ${targetActor.data.name}.<br>${flavor}
                            <p><em>Manually apply ${number} HP of healing to ${targetActor.data.name}</em></p>` });
                    else {
```

**Comment by Merudo** [comment]
Turns out it is an issue if any macro with charges is associated to a feat or item through the Dynamic Effects module.

Clicking the feat always depletes the item of one charge independently of any effect of the macro. Hence, why it wasted one charge of Lay on Hands was wasted.

**Pull request merged**